### PR TITLE
TST: Insert 'match' to bare pytest raises

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref #30999
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/test_errors.py [hunk 1]
```diff
@@ -22,12 +22,12 @@
...
 
-    e = getattr(errors, exc)
-    assert e is not None
+    err = getattr(errors, exc)
+    assert err is not None
 
     # check that we can raise on them
-    with pytest.raises(e):
-        raise e()
+    with pytest.raises(err, match=""):
```
Comment (simonjayhawkins) [inline pandas/tests/test_errors.py]: the match argument is treated as a regex, did you mean "^$", i.e. no message?

File: pandas/tests/test_take.py [hunk 2]
```diff
@@ -446,7 +451,11 @@ def test_take_empty(self, allow_fill):
         result = algos.take(arr, [], allow_fill=allow_fill)
         tm.assert_numpy_array_equal(arr, result)
 
-        with pytest.raises(IndexError):
+        msg = (
+            r"cannot do a non-empty take from an empty axes.|"
+            r"indices are out-of-bounds"
```
Comment (simonjayhawkins) [inline pandas/tests/test_take.py]:
```suggestion
            "indices are out-of-bounds"
```

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @MomIsBestFriend for the PR.

File: pandas/tests/test_errors.py (hunk 1 above)
Comment (MomIsBestFriend) [reply pandas/tests/test_errors.py]: Correct, will fix.

(Thank you)

**Comment by MomIsBestFriend** [comment]
@simonjayhawkins Do you mind if I open a master issue, instead of #23922 ?
There's a lot of noise there about how to find a bare pytest raises, I have a test case and I can point at the line number at each file with a bare pytest raise, and will make a list of occurrences (similar to #29886)

---

Do you think it's a good idea?

**Comment by simonjayhawkins** [comment]
> Do you think it's a good idea?

The format of #29886 worked well for community engagement. so OK with opening a new issue to supersede #23922.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MomIsBestFriend lgtm pending green.

File: pandas/tests/test_lib.py [hunk 3]
```diff
@@ -100,9 +101,14 @@ def test_maybe_indices_to_slice_right_edge(self):
         assert not isinstance(maybe_slice, slice)
         tm.assert_numpy_array_equal(maybe_slice, indices)
 
-        with pytest.raises(IndexError):
+        msg = (
+            "index 100 is out of bounds for axis 0 with size 100|"
```
Comment (WillAyd) [inline pandas/tests/test_lib.py]: minor but could simplify with a group or even character class

```suggestion
            "index 100 is out of bounds for axis (0|1) with size 100|"
```

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/test_downstream.py [hunk 4]
```diff
@@ -136,7 +136,12 @@ def test_missing_required_dependency():
     # https://github.com/MacPython/pandas-wheels/pull/50
     call = ["python", "-sSE", "-c", "import pandas"]
 
-    with pytest.raises(subprocess.CalledProcessError) as exc:
+    msg = (
+        r"Command '\['python', '-sSE', '-c', 'import pandas'\]' "
+        r"returned non-zero exit status 1."
```
Comment (WillAyd) [inline pandas/tests/test_downstream.py]:
```suggestion
        "returned non-zero exit status 1."
```

I don't think need the raw prefix here either

**Comment by jbrockmendel** [comment]
pending addressing comments from @WillAyd, LGTM

File: pandas/tests/test_lib.py (hunk 3 above)
Comment (MomIsBestFriend) [reply pandas/tests/test_lib.py]: That's really smart!

**Comment by simonjayhawkins** [comment]
Thanks @MomIsBestFriend

**Pull request merged**